client/src/main/java/com/paypal/selion/platform/grid/SeleniumGridListener.java

```diff
@@ -137,10 +137,13 @@ public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
                     System.exit(1); // NOSONAR
                 }
             }
+        } catch (RuntimeException e) { // NOSONAR
+            Grid.getThreadLocalException().set(e);
         } catch (Exception e) { // NOSONAR
-            logger.log(Level.WARNING, "An error occurred while processing beforeInvocation: " + e.getMessage(), e);
+            //Must wrap the Exception to RuntimeException, So that users no need to
+            //add try catch block in their test case when invoke Grid.driver()
+            Grid.getThreadLocalException().set(new RuntimeException(e.getMessage(), e));
```

Comment:
@sebady 
a) Why not convert to RuntimeException at the throw site instead?
     Purpose is to catch all the exception, should not leak any exception. Already code validation and all the custom exceptions are RuntimeException only.

b) Also what if multiple exceptions have occurred in BeforeX config methods before Grid.driver(). We are saving only the last and clobbering all earlier ones.?
       Scenario1: Our logic in SeleniumGridListener.beforeInvocation() wont execute at all for the BeforeX config methods in Non session sharing test. So that no exception will be occurred in belforeInvocation method. If some code problem in BeforeX methods then subsequent test method will be skipped.
       Scenario2: Logic in SeleniumGridListener.beforeInvocation() method will be executed for BeforeClass method in Session sharing test. If something went wrong then exception will be saved and will be thrown when Grid.driver() invoked in BeforeClass method itself. So subsequent test case will be skipped. 
	   
I have tried to reproduce the scenario you are talking about here. But I am not able to reproduce this issue by having valid scenario. In case more than one issues are occurred then latest exception only thrown to the user. Once user solve the latest issue then user will get previous exception for the second run. Is it fine?